src/components/localization/language-switcher.js

```diff
@@ -31,12 +31,36 @@ class LanguageSwitch extends Component {
 
     handleSelect = e => {
         const { value } = e.target
+
         const path = value === '/en' ? '/' : value
         if (path === '/ach') {
             localStorage.setItem('i18n', 'ach')
             window.location.href = '/ach'
         } else {
-            navigate(path, { hrefLang: path })
+            /*
+                can be something like /es/about/
+                or just /about/
+                or just /
+            
+            */
+            const current_path = window.location.pathname
+            let destination_path = path
+
+            const current_lang = localStorage.getItem('i18n') || 'en'
+            if (current_lang === 'en') {
```

Comment:
```suggestion
            if (value === '/en') {
```

Reply:
value is the destination path, meanwhile, we want to know the what is the current url path

Reply:
line 47 value={`/${this.state.language}`}